tensorflow_model_optimization/python/core/sparsity_tf2/pruner.py

```diff
@@ -211,4 +211,64 @@ def prune(self, optimizer, var, grad):
     mask = optimizer.get_slot(var, 'mask')
     threshold = optimizer.get_slot(var, 'threshold')
     self.update_masks([(var, mask, threshold)], step=optimizer.iterations)
-    self._mask_weight(var, mask)
+    self._apply_mask(var, mask)
+
+class LTHPruner(Pruner):
+  """
+  Implementation of Lottery Ticket Hypothesis experiments.
+  """
+
+  def __init__(self,
+      pruning_schedule=pruning_sched.ConstantSparsity(0.5, 0),
+      save_schedule=None,
+      block_size=(1,1),
+      block_pooling_type='AVG',
+  ):
+    """The logic for magnitude-based pruning weight tensors.
```

Comment:
Will update. Aren't docstrings usually within each method definition?

Reply:
Sorry, brain was dead when I wrote this comment.
You're right, in python docstrings go inside the object you're documenting.

If it's a docstring for the constructor though it can either go in your class docstring or your __init__ method docstring. I'm okay w/ it being either one, so this is fine.